### PR TITLE
Cleanup and composability

### PR DESCRIPTION
**Pull request by clalancette**
This PR does a bit of cleanup (mostly re-enabling cppcheck), and makes the teleop_twist_joy code composable.  @cottsay, review appreciated.

### CODE REVIEW

File: src/teleop_twist_joy.cpp [hunk 1]
```diff
@@ -216,3 +219,5 @@ void TeleopTwistJoy::Impl::joyCallback(const sensor_msgs::msg::Joy::SharedPtr jo
 }
 
 }  // namespace teleop_twist_joy
+
+RCLCPP_COMPONENTS_REGISTER_NODE(teleop_twist_joy::TeleopTwistJoy)
```
Comment (mjcarroll) [inline src/teleop_twist_joy.cpp:223]: We've been putting the components include right above this line, but I don't think that it matters much.

**Review by mjcarroll — APPROVED** [review]
Small nit, but LGTM.

**Review by cottsay — APPROVED** [review]

File: src/teleop_twist_joy.cpp (hunk 1 above)
Comment (clalancette) [reply src/teleop_twist_joy.cpp:223]: Yeah, it makes copy-n-paste easier.  But I don't really love putting a random `#include` at the end of source code, so I think I'll leave it as-is for now.

**Comment by clalancette** [comment]
Thanks for the reviews!  Merging.

**Pull request merged**